Tests/Services/NfieldSurveyLanguageTranslationTests.cs

```diff
@@ -0,0 +1,206 @@
+        public void TestQueryAsync_SurveyIdIsNull_ThrowsArgumentNullException()
+        {
+            var target = new NfieldSurveyLanguageTranslationsService();
+            Assert.Throws<ArgumentNullException>(() => UnwrapAggregateException(target.QueryAsync(null)));
+        }
+
+        [Fact]
+        public void TestQueryAsync_SurveyIdIsEmpty_ThrowsArgumentException()
+        {
+            var target = new NfieldSurveyLanguageTranslationsService();
+            Assert.Throws<ArgumentException>(() => UnwrapAggregateException(target.QueryAsync(string.Empty)));
+        }
+
+        [Fact]
+        public void TestQueryAsync_ServerReturnsQuery_ReturnsListWithLanguages()
+        {
+            var expectedTranslations = new Language[]
+            {
+                new Language{Name = "X", Id = 1},
+                new Language{Name = "Y", Id = 2}
+            };
+
+            var mockedNfieldConnection = new Mock<INfieldConnectionClient>();
+            var mockedHttpClient = CreateHttpClientMock(mockedNfieldConnection);
+            mockedHttpClient
+                .Setup(client => client.GetAsync(new Uri(ServiceAddress, $"Surveys/{SurveyId}/LanguageTranslations")))
+                .Returns(CreateTask(HttpStatusCode.OK, new StringContent(JsonConvert.SerializeObject(expectedTranslations))));
+
+            var target = new NfieldSurveyLanguageTranslationsService();
+            target.InitializeNfieldConnection(mockedNfieldConnection.Object);
+
+            var actualTranslations = target.QueryAsync(SurveyId).Result;
+            Assert.Equal(expectedTranslations[0].Name, actualTranslations.ToArray()[0].Name);
+            Assert.Equal(expectedTranslations[0].Id, actualTranslations.ToArray()[0].Id);
+            Assert.Equal(expectedTranslations[1].Name, actualTranslations.ToArray()[1].Name);
+            Assert.Equal(expectedTranslations[1].Id, actualTranslations.ToArray()[1].Id);
+            Assert.Equal(2, actualTranslations.Count());
+        }
+
+        #endregion
+
+        #region AddAsync
+
+        [Fact]
+        public void TestAddAsync_SurveyIdIsNull_ThrowsArgumentNullException()
+        {
+            var target = new NfieldSurveyLanguageTranslationsService();
+            Assert.Throws<ArgumentNullException>(() => UnwrapAggregateException(target.AddAsync(null, new SurveyLanguageTranslations())));
+        }
+
+        [Fact]
+        public void TestAddAsync_SurveyIdIsEmpty_ThrowsArgumentException()
+        {
+            var target = new NfieldSurveyLanguageTranslationsService();
+            Assert.Throws<ArgumentException>(() => UnwrapAggregateException(target.AddAsync(string.Empty, new SurveyLanguageTranslations())));
+        }
+
+        [Fact]
+        public void TestAddAsync_ServerAcceptsLanguageTranslations_ReturnsLanguageTranslations()
+        {
+            var translations = new SurveyLanguageTranslations()
+            {
+                ButtonNext = "Next",
+                ButtonBack = "Back",
```

Comment:
I don't know if it would be necessary add all the properties to test if the SerializeObject & DeserializeObject works for all

Reply:
There are many more properties, I think testing one or two is enough - we should not have to test whether (de)serialization works as it should. Or are you expecting issues with it?